datadog_checks_base/datadog_checks/base/utils/db/query.py

```diff
@@ -26,6 +26,12 @@ def __init__(self, query_data):
         self.extras = None
         self.tags = None
 
+    def copy(self):
```

Comment:
By copying the query data, the original query is never modified/compiled? So do you ever use the original query anymore?

Reply:
The original payload data is never modified. The original query data is used to create new Query objects, but the existing `Query` constant objects are never modified/compiled (because they have to be compiled for each check instance).
This is a bit hacky and that's why #7750 is a bit cleaner.

Reply:
Ah I see now